ocaml/xapi/server_helpers.ml

```diff
@@ -117,44 +124,52 @@ let dispatch_exn_wrapper f =
     f()
   with exn -> let code, params = ExnHelper.error_of_exn exn in API.response_of_failure code params
 
-let do_dispatch ?session_id ?forward_op ?self called_async supports_async called_fn_name op_fn
-    marshaller_fn fd http_req label generate_task_for =
+let do_dispatch ?session_id ?forward_op ?self supports_async called_fn_name op_fn
+    marshaller fd http_req label sync_ty generate_task_for =
```

Comment:
is there a reason why the parameter position of "async-ness" was moved?

Reply:
yes - `sync_ty` takes values of `Sync, Async, InternalAsync`, so we would be duplicating data by keeping the other async variable around

Reply:
I understand that, the question is why is its position further back.